external/fv3fit/fv3fit/reservoir/autoencoder.py

```diff
@@ -26,20 +41,195 @@ def encode(self, x: Union[np.ndarray, tf.Tensor]) -> np.ndarray:
     def decode(self, latent_x: Union[np.ndarray, tf.Tensor]) -> np.ndarray:
         return self.decoder.predict(latent_x)
 
+
+@io.register("dense-autoencoder")
+class AutoencoderPredictor(Reloadable):
+    _ENCODER_NAME = "encoder.tf"
+    _DECODER_NAME = "decoder.tf"
+    _CONFIG_FILENAME = "autoencoder_config.yaml"
+
+    def __init__(
+        self,
+        input_variables: Iterable[Hashable],
+        output_variables: Iterable[Hashable],
+        model: Autoencoder,
+    ):
+        self.input_variables = input_variables
+        self.output_variables = output_variables
+        self.model = model
+
+    def predict(self, X: np.ndarray):
+        return self.model.predict(X)
+
+    def dump(self, path: str) -> None:
+        with put_dir(path) as path:
+            encoder_filename = os.path.join(path, self._ENCODER_NAME)
+            self.model.encoder.save(encoder_filename)
+            decoder_filename = os.path.join(path, self._DECODER_NAME)
+            self.model.decoder.save(decoder_filename)
+            with open(os.path.join(path, self._CONFIG_FILENAME), "w") as f:
+                f.write(
+                    yaml.dump(
+                        {
+                            "input_variables": self.input_variables,
+                            "output_variables": self.output_variables,
+                        }
+                    )
+                )
+
     @classmethod
-    def load(cls, path: str) -> "Autoencoder":
+    def load(cls, path: str) -> "AutoencoderPredictor":
         with get_dir(path) as model_path:
             encoder = tf.keras.models.load_model(
                 os.path.join(model_path, cls._ENCODER_NAME)
             )
             decoder = tf.keras.models.load_model(
                 os.path.join(model_path, cls._DECODER_NAME)
             )
-        return cls(encoder=encoder, decoder=decoder)
+            with open(os.path.join(path, cls._CONFIG_FILENAME), "r") as f:
+                config = yaml.load(f, Loader=yaml.Loader)
+        autoencoder = Autoencoder(encoder=encoder, decoder=decoder)
+        return cls(
+            input_variables=config["input_variables"],
+            output_variables=config["output_variables"],
+            model=autoencoder,
+        )
 
-    def dump(self, path: str) -> None:
-        with put_dir(path) as path:
-            encoder_filename = os.path.join(path, self._ENCODER_NAME)
-            self.encoder.save(encoder_filename)
-            decoder_filename = os.path.join(path, self._DECODER_NAME)
-            self.decoder.save(decoder_filename)
+
+@dataclasses.dataclass
+class DenseAutoencoderHyperparameters(Hyperparameters):
+    """
+    input_variables: variables to encode/decode
+    output_variables: variables to encode/decode, must match
+        input_variables
```

Comment:
There's no requirement that a `Hyperparameters` class has `input_variables` and `output_variables` attributes, if the inputs and outputs must be the same please use a single attribute like `state_variables`.